Observer/Model/ProcessOrder/CardsMetadata.php

```diff
@@ -91,14 +102,17 @@ public function execute(Observer $observer)
 
     /**
      * @param array $data
-     * @param array $payment
+     * @param array $paymentData
      * @param string $key
+     * @param Payment $payment
      * @return void
+     * @throws LocalizedException
      */
-    private function populateCardData(array &$data, array $payment, string $key): void
+    private function populateCardData(array &$data, array $paymentData, string $key, Payment $payment): void
```

Comment:
`$order->getPayment();` returns `OrderPaymentInterface|null` and we accept `Payment` in the fourth param. Wonder if it will cause issues later if its a different implementation that is returned.